tests/common/include/tests/common/benchmark_find_volume.inl

```diff
@@ -22,9 +24,12 @@ unsigned int gbench_repetitions = DETRAY_BENCHMARKS_REP;
 unsigned int gbench_repetitions = 0;
 #endif
 
+vecmem::host_memory_resource host_mr;
```

Comment:
If possible, this should go to the read detector function

Reply:
Indeed, it was in the read detector function in the earlier commits; and the destructor of vector gave the seg fault error because the memory resource does not persist out of the function. That's why it is out of the function now

Reply:
Ah, I see. Well if we cannot move it, then lets leave it here

Reply:
Memory resource objects should be constructed to have long lifetimes indeed. They need to persist for the lifetime of every container that was created with their help. So definitely don't try to hide them too much, only creating them inside of "smaller" functions.